### PR TITLE
Define basic transliterator interface

### PR DESCRIPTION
**Pull request by frombrest**
Part of #1

### CODE REVIEW

**Comment by nbeloglazov** [comment]
High-level directory structure comment. Given that we are going to start with a single repo which will contain both tests, python library and potentially other libraries I was thinking about separating libraries by directories:

* tests
* python
* (potentially) csharp

The `tests` directory will contain only json files without any python code. To keep it language-agnostic.

File: tests/cyr_narcom_to_cyr/test_cases/test_1.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "tests": [
+    {
+      "name": "test_1",
+      "input": "FrGpMnWmcTq3",
+      "expected_output": "frgpmnwmctq3"
+    },
```
Comment (nbeloglazov) [inline tests/cyr_narcom_to_cyr/test_cases/test_1.json:7]: Regarding test structure. It seems like we'll need to support at least 3 different types of text: narkam, tarashk and lacinka. What do you think if test case contains potentially all three? For example:

```json
{
  "name": "some name",
  "cyr_nar": "снег",
  "cyr_taras": "сьнег",
  "lac": "śneh",
}
```

And then libraries can decide what they want to test. Or probably test multiple variations for each test.

File: python/setup.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python
+from setuptools import setup
+
+setup(
+    name="translit",
```
Comment (nbeloglazov) [inline python/setup.py:5]: Is this the name of the package under which it will be published? We should spend some time bikeshedding the name. Filed a bug to discuss it: https://github.com/belaudiobooks/lacinka/issues/4 No need to do anything here.

File: python/translit/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from .engine import convert
```
Comment (nbeloglazov) [inline python/translit/__init__.py:1]: Curious, why is this required? Seems like a no-op to me.

File: python/tests/test_cyr_nar_to_cyr_taras.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import pytest
+import glob
+
+from translit import convert
+from translit.cases import Case
+
+
+def read_json_data():
+    test_file_list = glob.glob('../tests/test_*.json')
+    all_test_cases = []
```
Comment (nbeloglazov) [inline python/tests/test_cyr_nar_to_cyr_taras.py:11]: How about making it a dict from filename to list of tests? Or alternatively marking each test with filename that it came from? During test failure it would be good not only print the test name, but also the filename it came from for easier debugging.

File: python/translit/cases.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+"""
+Class mimic to enum to keep consistent definition
+all possible transliteration case options.
+"""
+
+
+class Case:
```
Comment (nbeloglazov) [inline python/translit/cases.py:7]: We should discuss with Vincuk what's the best name for this enum. Is it  alphabet or orthography or something else. No need to do anything here right now.

File: python/translit/converters/translit.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    def convert(self, text):
+        """
+        Function responsible for conversion
+        :param text: String text to convert
+        :return: String conversion result
+        """
+        raise NotImplementedError("Need to be implemented")
+
+    def meta(self):
```
Comment (nbeloglazov) [inline python/translit/converters/translit.py:16]: That's a cool idea to have function like that.

File: python/translit/engine.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+Converter map contains lambdas to do not create 
+all objects during map initialization
+"""
+converters = {
+    Case.CYR_NAR + "_TO_" + Case.CYR_TARAS: lambda: CyrNarToCyrTaras(),
+    Case.CYR_NAR + "_TO_" + Case.LAT: lambda: CyrNarToLat(),
+    Case.CYR_TARAS + "_TO_" + Case.LAT: lambda: CyrTarasToLat(),
+    Case.CYR_TARAS + "_TO_" + Case.CYR_NAR: lambda: CyrTarasToCyrNar(),
+    Case.LAT + "_TO_" + Case.CYR_TARAS: lambda: LatToCyrTaras(),
+    Case.LAT + "_TO_" + Case.CYR_TARAS: lambda: LatToCyrNar()
```
Comment (nbeloglazov) [inline python/translit/engine.py:19]: Let's start with a single coverter for now. And add more later once we flesh out structure of the first one. We can start with TARAS => LAT.

File: python/translit/engine.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+
+def convert(text, source_case, target_case):
```
Comment (nbeloglazov) [inline python/translit/engine.py:23]: Do call meta() function anywhere? Doesn't looks like it's exposed at all to the user.

**Review by nbeloglazov — CHANGES_REQUESTED** [review]

File: python/translit/__init__.py (hunk 3 above)
Comment (frombrest) [reply python/translit/__init__.py:1]: later you will be able to import this function by this statement:
```
from translit import convert
```

without this line you will need to do:
```
from translit.engine import convert
```

File: python/translit/converters/translit.py (hunk 6 above)
Comment (frombrest) [reply python/translit/converters/translit.py:16]: To be honest, I am not sure we really need class based implementation.
May be better to have dict key: Conversion direction and value: reference to function? 
What do you think @nbeloglazov ?

Because you are right here: https://github.com/belaudiobooks/lacinka/pull/2#discussion_r1264603385
I do not use this function.

File: python/translit/engine.py (hunk 8 above)
Comment (frombrest) [reply python/translit/engine.py:23]: https://github.com/belaudiobooks/lacinka/pull/2#discussion_r1264609464

File: python/translit/converters/translit.py (hunk 6 above)
Comment (nbeloglazov) [reply python/translit/converters/translit.py:16]: Yeah. Starting with individual functions sounds good. Later should be trivial enough to refactor to interface-like approach as long as it doesn't affect public API.

File: python/translit/cases.py (hunk 5 above)
Comment (frombrest) [reply python/translit/cases.py:7]: Let's maybe create another issue for that?

File: python/tests/test_cyr_nar_to_cyr_taras.py (hunk 4 above)
Comment (frombrest) [reply python/tests/test_cyr_nar_to_cyr_taras.py]: Fixed image

**Review by nbeloglazov — APPROVED** [review]
